packages/contracts/contracts/interfaces/v0.8.x/IClaimMinter.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+pragma solidity ^0.8.0;
+
+/**
+ * @title IClaimMinter
+ * @author Art Blocks Inc.
+ * @notice Interface for the ClaimMinter contract that enables admin to pre-mint tokens
+ * for a project and allows collectors to claim tokens in an arbitrary order with
+ * configurable pricing.
+ */
+interface IClaimMinter {
+    // Events
+    event PriceConfigured(uint256 basePriceInWei, uint256 priceIncrementInWei);
+    event TimestampStartConfigured(uint256 timestampStart);
+    event TokensPreMinted(uint256 amount);
+    event TokenClaimed(uint256 tokenId, address claimant, uint256 price);
+
+    /**
+     * @notice Returns all claimed bitmaps
+     * @return An array of all claimed bitmaps
+     */
+    function getAllClaimedBitmaps() external view returns (string memory);
+
+    /**
+     * @notice Sets the base price in wei for token 0 and the price increment in wei.
+     * Only callable by the core contract's Admin ACL.
+     * @param basePriceInWei_ Base price in wei for token 0.
+     * @param priceIncrementInWei_ Price increment in wei.
+     */
+    function configurePricePerTokenInWei(
+        uint256 basePriceInWei_,
+        uint256 priceIncrementInWei_
+    ) external;
+
+    /**
+     * @notice Sets the timestamp when claiming can begin.
+     * Only callable by the core contract's Admin ACL.
+     * @param timestampStart_ Unix timestamp when claiming can begin.
+     */
+    function configureTimestampStart(uint256 timestampStart_) external;
+
+    /**
+     * @notice Pre-mints `amount` tokens to this contract.
+     * @param amount Number of tokens to pre-mint.
+     */
+    function preMint(uint256 amount) external;
+
+    /**
+     * @notice Claims token `tokenId` by paying the required price.
+     * @param tokenId Token ID to claim.
+     */
+    function claimToken(uint256 tokenId) external payable;
+
+    /**
+     * @notice Checks if a token is claimed using bitmap storage
+     * @param tokenId The token ID
+     * @return True if the token is claimed
+     */
+    function isTokenClaimed(uint256 tokenId) external view returns (bool);
```

Comment:
Since project ID is an immutable constant for this minter contract, I think it's cleaner to have the input here be tokenNumber instead of tokenId (e.g. 3 instead of 501000003).
We can then check that against max invocations, and build tokenId from the project id + token number.
Reasoning here is to avoid any potential bugs/vulnerabilities that we may not be thinking of if users were to input a weird tokenId (even things like overflow/underflow, even though we would technically expect those to revert in sol 0.8).